test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs

```diff
@@ -0,0 +1,866 @@
+
+            project.EmbedInteropTypes.Should().BeTrue();
+        }
+
+        [Fact]
+        public void It_does_not_add_commands_when_commands_is_not_set_in_the_ProjectJson()
+        {
+            _emptyProject.Commands.Should().BeEmpty();
+        }
+
+        [Fact]
+        public void It_does_not_add_commands_when_commands_is_not_a_JsonObject()
+        {
+            var json = new JObject();
+            json.Add("commands", true);
+            var project = GetProject(json);
+
+            project.Commands.Should().BeEmpty();
+        }
+
+        [Fact]
+        public void It_does_not_add_the_commands_when_its_value_is_not_a_string()
+        {
+            var json = new JObject();
+            var commands = new JObject();
+            json.Add("commands", commands);
+
+            commands.Add("commandKey1", "commandValue1");
+            commands.Add("commandKey2", true);
+
+            var project = GetProject(json);
+
+            project.Commands.Count.Should().Be(1);
+            project.Commands.First().Key.Should().Be("commandKey1");
+            project.Commands.First().Value.Should().Be("commandValue1");
+        }
+
+        [Fact]
+        public void It_does_not_add_scripts_when_scripts_is_not_set_in_the_ProjectJson()
+        {
+            _emptyProject.Scripts.Should().BeEmpty();
+        }
+
+        [Fact]
+        public void It_does_not_add_scripts_when_scripts_is_not_a_JsonObject()
+        {
+            var json = new JObject();
+            json.Add("scripts", true);
+            var project = GetProject(json);
+
+            project.Scripts.Should().BeEmpty();
+        }
+
+        [Fact]
+        public void It_does_not_add_the_scripts_when_its_value_is_neither_a_string_nor_an_array_of_strings()
+        {
+            var scriptArrayValues = new [] {"scriptValue2", "scriptValue3"};
+
+            var json = new JObject();
+            var scripts = new JObject();
+            json.Add("scripts", scripts);
+
+            scripts.Add("scriptKey1", "scriptValue1");
+            scripts.Add("scriptKey3", new JArray(scriptArrayValues));
+
+            var project = GetProject(json);
+
+            project.Scripts.Count.Should().Be(2);
+            project.Scripts.First().Key.Should().Be("scriptKey1");
+            project.Scripts.First().Value.Should().Contain("scriptValue1");
+            project.Scripts["scriptKey3"].Should().Contain(scriptArrayValues);
+        }
+
+        [Fact]
+        public void It_throws_when_the_value_of_a_script_is_neither_a_string_nor_array_of_strings()
+        {
+            var json = new JObject();
+            var scripts = new JObject();
+            json.Add("scripts", scripts);
+
+            scripts.Add("scriptKey2", true);
+
+            Action action = () => GetProject(json);
+
+            action.ShouldThrow<FileFormatException>()
+                .WithMessage("The value of a script in project.json can only be a string or an array of strings");
```

Comment:
This is inline with comments above. Why are we accepting invalid input in some scenarios but throwing in others?